tests/filesystem/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+#![no_std]
+#![feature(const_str_as_bytes)]
+#![feature(const_raw_ptr_deref)]
+
+extern crate alloc;
+
+use linux_kernel_module::filesystem::{self, FileSystem, FileSystemFlags};
+use linux_kernel_module::{self, CStr};
+
+struct TestFSModule {
+    _fs_registration: filesystem::Registration<TestFS>,
+}
+
+// This is safe because Registration doesn't actually expose any methods.
+unsafe impl Sync for TestFSModule {}
```

Comment:
`Registration` ought to be `Sync` then. As a general rule, modules shouldn't need `unsafe` on account of our APIs.

Reply:
I see. I can add 

```rust
unsafe impl<T> Sync for Registration<T> where T: FileSystem {}
```

to `filesystem.rs` but I'm not sure whether this is the right thing to do. Couldn't someone who creates `Registration<MyFS>` (where `MyFS` implements `Filesystem`) define methods on this concrete type? This would break the assumption that there are no methods defined for `Registration<T: Filesystem>` and therefore the `Sync` marker for it would no longer be legal.

Reply:
Anyone implementing methods on `Registration<T>` via trait or something would need to keep the invariants -- that's always the case though. You can see a similar `impl` in `chrdev.rs`.

We do need to mark `FileSystem` as `: Sync` though.